### PR TITLE
Finish GitHub org move

### PR DESCRIPTION
**Pull request by srenatus**
As mentioned in #1281, this seems necessary to finish the move.

However, we probably shouldn't merge this until Travis has been updated -- I'm lacking authorization for that, but I've requested it (see #1280).

With this branch, `docker build -t test .` looks successful; and `make test` passes locally ✔️

### CODE REVIEW

**Comment by ericchiang** [comment]
@rithujohn191 can you make me an owner of the dexidp org? I need that to re-enable travis

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ go:
   - 1.9
   - '1.10'
 
-go_import_path: github.com/coreos/dex
```
Comment (ericchiang) [inline .travis.yml]: In hindsight, this should let us forgo the package rename. We might be able to split out removing the bill of materials.

Comment (srenatus) [reply .travis.yml]: Technically, yeah... but if someone wanted to depend on post-move dex, those names would have to match, wouldn't they? (Ignoring the new go modules stuff for a moment.)

Comment (ericchiang) [reply .travis.yml]: I would feel really bad if someone was importing dex 😬 we don't maintain these packages for general use.

Comment (srenatus) [reply .travis.yml]: 😄 It's quite convenient to have an all-in-memory-dex in tests. We do it in a few places... 🤔

Comment (srenatus) [reply .travis.yml]: I suppose the same holds for using its grpc client definitions.

**Comment by rithujohn191** [comment]
@ericchiang changed your role to owner

**Comment by ericchiang** [comment]
(closing and re-opening to see if this kicks travis)

**Comment by srenatus** [comment]
@ericchiang Ok, nuked the bill of materials, and updated the quay references anticipating that we  publish this into `quay.io/dexidp` (which I get a 403 on right now?), #1284.

I'd really rather have the package imports match the repository. Even if we want no one to import dex, and document how to generate the gRPC client package yourself, it feels like an unnecessary extra step for contributors to ensure that the checkout matches a different import path (like, "clone this into `${GOPATH}/coreos/dex`"). While doing a rename now might necessitate a few fixes in open PRs, on the long run, _renaming-all-the-things now_ seems like the simpler solution to me. What do you think?

File: Dockerfile [hunk 2]
```diff
@@ -2,8 +2,8 @@ FROM golang:1.10.2-alpine
 
 RUN apk add --no-cache --update alpine-sdk
 
-COPY . /go/src/github.com/coreos/dex
-RUN cd /go/src/github.com/coreos/dex && make release-binary
+COPY . /go/src/github.com/dexidp/dex
```
Comment (ericchiang) [inline Dockerfile]: Probably early to update the quay references. Let's do this once we get that account setup.

I'll need your quay username too.

Comment (ericchiang) [reply Dockerfile]: added

**Comment by srenatus** [comment]
@ericchiang ✔️ squashed

**Pull request merged**